### PR TITLE
DT-2229 refactor

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/AmazonSqsFactory.kt [hunk 1]
```diff
@@ -4,30 +4,60 @@ import com.amazonaws.auth.AWSStaticCredentialsProvider
...
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
 
 class AmazonSqsFactory {
+  companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
 
-  fun awsAmazonSQS(accessKeyId: String, secretAccessKey: String, region: String) =
+  fun awsSqsClient(queueName: String, accessKeyId: String, secretAccessKey: String, region: String, asyncClient: Boolean = false): AmazonSQS =
```
Comment (mikehalmamoj) [inline lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/AmazonSqsFactory.kt:18]: Moved from HmppsQueueFactory

File: lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package uk.gov.justice.hmpps.sqs
+
+import org.springframework.boot.context.properties.ConfigurationProperties
+import org.springframework.boot.context.properties.ConstructorBinding
+
+@ConstructorBinding
+@ConfigurationProperties(prefix = "hmpps.sqs")
+data class HmppsSqsProperties(
```
Comment (mikehalmamoj) [inline lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt:8]: Renamed class to match prefix hmpps.sqs

File: lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsConfiguration.kt [hunk 3]
```diff
@@ -9,9 +9,9 @@ import org.springframework.context.annotation.DependsOn
 import org.springframework.jms.annotation.EnableJms
 
 @Configuration
-@EnableConfigurationProperties(HmppsQueueProperties::class)
+@EnableConfigurationProperties(HmppsSqsProperties::class)
 @EnableJms
-class HmppsQueueConfiguration {
+class HmppsSqsConfiguration {
```
Comment (mikehalmamoj) [inline lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsConfiguration.kt:14]: Renamed class to match library name.  However all components that trade in HmppsQueue instances are still prefixed HmppsQueue.

File: lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    val queueAccessKeyId: String = "",
+    val queueSecretAccessKey: String = "",
+    val asyncQueueClient: Boolean = false,
+    val dlqName: String,
+    val dlqAccessKeyId: String = "",
+    val dlqSecretAccessKey: String = "",
+    val asyncDlqClient: Boolean = false,
+  )
+
+  init {
```
Comment (mikehalmamoj) [inline lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt:25]: Added some validation to catch bad configuration.

File: lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    queues.forEach { (queueId, queueConfig) ->
+      if (queueId != queueId.lowercase()) throw InvalidHmppsSqsPropertiesException("queueId $queueId is not lowercase")
```
Comment (mikehalmamoj) [inline lib/src/main/kotlin/uk/gov/justice/hmpps/sqs/HmppsSqsProperties.kt:27]: I discovered that Spring cannot unwrap environment variables of the format `HMPPS_SQS_QUEUES_QUEUE_NAME_QUEUE_ACCESS_KEY_ID" into the queues Map in this class, so the queueId has to be lower case.

File: test-app/src/main/resources/application.yml [hunk 6]
```diff
@@ -59,20 +59,3 @@ management:
 oauth:
   endpoint.url: http://localhost:8090/auth
 
-# These should not be required in your main application and should instead be provided by helm values referencing Kuberentes secrets - these are only here for unit tests
```
Comment (mikehalmamoj) [inline test-app/src/main/resources/application.yml:62]: Removed the unit test as the test-app is a kind of living example application but in real consumers of the library there should not be any config required in application.yml

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**